CHANGELOG.md

```diff
@@ -1,5 +1,337 @@
+
+  [AaronO]: https://github.com/AaronO
+  [pull/225]: https://github.com/cloudflare/wrangler/pull/225
+
+  - **Don't break `wrangler preview` when kv namespaces present - [ashleymichal], [pull/353]**
+
+  This PR fixes a critical bug where using `wrangler preview` on a project with [Workers KV][kv] namespaces causes the command to throw an error.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/353]: https://github.com/cloudflare/wrangler/pull/353
+
+  - **feat kv config - [ashleymichal], [pull/333]**
+
+  If you've tried to use [Workers KV][kv] in Wrangler, you've probably had a bad time! This PR, along with #334, build support for handling and correctly uploading KV namespace information with your Wrangler project.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/333]: https://github.com/cloudflare/wrangler/pull/333
+  [kv]: https://www.cloudflare.com/products/workers-kv/
+
+  - **Always use multipart upload - [ashleymichal], [issue/280][pull/328]**
+
+  We've updated how Wrangler uploads projects to the Workers runtime - as Wrangler supports the entire Workers ecosystem (including tools like [Workers KV][kv], the publishing process should use [multipart forms](https://dev.to/sidthesloth92/understanding-html-form-encoding-url-encoded-and-multipart-forms-3lpa) to allow uploading all of the data that represents a Workers project in a single step.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [issue/280]: https://github.com/cloudflare/wrangler/issue/280
+  [pull/328]: https://github.com/cloudflare/wrangler/pull/328
+
+  - **Better error printing - [granjef3], [pull/327]**
+
+  We've updated how we log errors in Wrangler's output to make it a little easier to read. If you're a Rust developer interested in _how_ we did this, check out the [pull request][pull/327]!
+
+  [granjef3]: https://github.com/granjef3
+  [pull/327]: https://github.com/cloudflare/wrangler/pull/327
+
+  - **Add message module for terminal output - [ashleymical], [issue/219][pull/263]**
+
+    We've made improvements to Wrangler's terminal output functionality, with support for various log levels and implementations in Wrangler's API for easily using the log levels in future development.
+
+    The new terminal output functionality can be used by importing the `terminal::message` crate:
+
+    ```rust
+    use crate::terminal::message;
+
+    message::info("Building project") // "💁‍ Building project"
+    message::success("Your project has been deployed!") // "✨ Your project has been deployed!"
+
+    // Other available functions:
+    // message::warn, message::user_error, message::service_error, message::working, message::preview
+    ```
+
+    [ashleymical]: https://github.com/ashleymichal
+    [issue/219]: https://github.com/cloudflare/wrangler/issues/219
+    [pull/263]: https://github.com/cloudflare/wrangler/pull/263
+
+  - **Adds more descriptive subdomain errors - [EverlastingBugstopper], [issue/207][pull/259]**
```

Comment:
feature

Reply:
ah, this is in the feature section! is that right or did you want it somewhere else?

Reply:
yeah this is a feature i'm sorry my comment makes no sense